### PR TITLE
Add twilio-node typedefs

### PR DESCRIPTION
**Pull request by nickiannone**

### CODE REVIEW

File: twilio/twilio.d.ts [hunk 1]
```diff
@@ -0,0 +1,1408 @@
...
+    list: RestMethod;
+  }
+
+  /// resources/sip/Domains.js
+  export interface IPAccessControlListMappingInstance {
+    get: RestMethod;
+    delete: RestMethod;
+  }
+
+  export interface IPAccessControlListMappingResource {
```
Comment (sandersn) [inline twilio/twilio.d.ts]: seems like these could be a single `ListMappingResource<T>` and then have a bunch of type aliases like:

```ts
type IPAccessControlListMappingInstance = ListMappingResource<IPAccessControlMappingInstance>;
type CredentialListResource = ListMappingResource<CredentialListInstance>;
...
```

File: twilio/twilio.d.ts [hunk 2]
```diff
@@ -0,0 +1,1408 @@
...
+
+    get: RestMethod;
+    post: RestMethod;
+    create: RestMethod;
+    list: RestMethod;
+  }
+
+  /// resources/lookups/PhoneNumbers.js
+  export interface PhoneNumberInstance {
+    get: RestMethod;
```
Comment (sandersn) [inline twilio/twilio.d.ts]: a lot of these classes are structurally identical (PhoneNumberInstance, EventInstance, AlertInstance, ...) or very close (AlertResource, EventResource, CountryResource, ...). Is it really worthwhile to have so many different ones? The compiler will treat them all the same &mdash; you'll be able to assign an AlertInstance to a PhoneNumberInstance property, for example.

Comment (nickiannone) [inline twilio/twilio.d.ts]: I'm trying to work through these; my desired approach is to make a couple of base interfaces for each one with the minimal subset of methods and just patch in other methods/properties as needed (ie. `export interface ServiceResource extends ListMappingResource<ServiceInstance> {}` or `export interface UsageRecordResource extends BaseMappingResource<UsageRecordInstance> { get: RestMethod; daily: UsageRecordRange; /* ... */ }`). Would this be an acceptable compromise?

Comment (sandersn) [inline twilio/twilio.d.ts]: Yeah, good intellisense is a worthy goal. You just have to weigh it against the confusion people get when they can pass "different" types in by mistake. In this case I think you are right and that having representative names is the better choice.

Do note that `type` is just an alias and pre-2.0 will never be displayed by the compiler. (I think that changed right after the 2.0 beta started). So `interface X extends Base<Y>` is probably the safer pattern.

Comment (nickiannone) [inline twilio/twilio.d.ts]: I committed it using `type`, but I can change that back to using `interface` real quick. I'll do that and re-commit.

Comment (nickiannone) [inline twilio/twilio.d.ts]: Re-committed.

**Pull request merged**